viewing_party/party.py

```diff
@@ -1,23 +1,224 @@
 # ------------- WAVE 1 --------------------
 
+
+## Create a function that will create a movie function
+# with title, genre, and rating into a dictiionary
+# If there is no title, genre, or rating, return None
+# If any of the three variables is missing, return None
+
+from decimal import DivisionByZero
+from tests.test_constants import USER_DATA_2
+
+
 def create_movie(title, genre, rating):
-    pass
+
+    if not title or not genre or not rating:
+        return None
+    else: 
+        new_movie = {
+            "genre" : genre, 
+            "rating": rating,
+            "title" : title,
+        }
+    return new_movie
+
+
+## Create a function to track when movies are watched
+## use the new_movie list to generate how the function works
+## user_data is what has been "watched:"
+## updated_data = add_to_watched(user_data, movie)
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+## Create a function to track a watch list for the user
+## user_data = {"watchlist"}
+## updated_data = add_to_watchlist(user_data, movie)
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+## def a function that will call upon the watched and watchlist functions
+##  adjust the values of watched and watchlist into sets
+## use set functions to create a set that compares the similarities
+## return a set or dictionary that outputs the difference 
+## Return an error message if the movie is on neither list
+
+def watch_movie(user_data, title):  
+    for movies in user_data["watchlist"]:
+        if movies["title"] == title: 
+            user_data["watchlist"].remove(movies)
+            user_data["watched"].append(movies)

```

Comment:
Right now we continue iterating over the watchlist after we perform our swap. In most cases this isn't a big deal, but say we were dealing with lists of thousands of movies. We would probably want to exit the loop right after we perform the operation we want, or we'll continuing to loop over potentially hundreds of more titles! How would we be able to halt execution once we performed our swap?

Another consideration: we are removing a value from a list we are actively iterating over, so after the first removal, we skip a value because the way iterators work (recall our discussion on the "danger zone" from the roundtable on "Iterating over Data"). Given the data for our tests, we know this isn't a concern for this application, but what if we were working with data that had duplicate titles?